as/json.js

```diff
@@ -63,7 +64,7 @@ function send(endpoint, head, body, callback) {
     var self = this;
 
     var outreq = self.channel.request(self.reqOptions);
-    self.tchannelJSON.send(outreq, endpoint, head, body, callback);
+    self.tchannelJSON.send(outreq, endpoint, extend(head, self.reqOptions.baseAppHeaders), body, callback);
```

Comment:
please avoid the `extend` call unless it exists.

Also your extending in the wrong direction...